atc/metric/emitter/newrelic.go

```diff
@@ -54,64 +64,22 @@ func (config *NewRelicConfig) NewEmitter() (metric.Emitter, error) {
 	}
 
 	return &NewRelicEmitter{
-		client:     client,
-		url:        fmt.Sprintf("https://insights-collector.newrelic.com/v1/accounts/%s/events", config.AccountID),
-		apikey:     config.APIKey,
-		prefix:     config.ServicePrefix,
-		containers: new(stats),
-		volumes:    new(stats),
+		Client:              client,
+		Url:                 fmt.Sprintf("https://insights-collector.newrelic.com/v1/accounts/%s/events", config.AccountID),
+		apikey:              config.APIKey,
+		prefix:              config.ServicePrefix,
+		containers:          new(stats),
+		volumes:             new(stats),
+		BatchSize:           int(config.BatchSize),
+		BatchDuration:       config.BatchDuration,
+		CompressionDisabled: config.CompressionDisabled,
+		LastEmitTime:        time.Now(),
+		NewRelicBatch:       make([]NewRelicEvent, 0),
```

Comment:
Hey, it came to my mind that we **might** (in further iterations) leverage slices w/ a predefined capacity (so that we avoid having to allocate memory on demand as this increases in the `append(`s we perform 🤔  

The only concern I'd have is with the subsequent slice reset on `emitBatch`, where we'd need to do something like `batch = batch[:0]` to keep the capacity, but I'm not sure if that's "leaky" (w/ regards to memory) by default or not hmmm

(ps.: just a comment, I don't think we need that right now 😁 )